crates/valence_entity/src/active_status_effects.rs

```diff
@@ -0,0 +1,148 @@
+use std::collections::HashMap;
+
+use bevy_ecs::prelude::*;
+
+use crate::status_effects::StatusEffect;
+
+/// [`Component`] that stores the [`ActiveStatusEffect`]s of an [`Entity`].
+#[derive(Component, Default, Debug)]
+pub struct ActiveStatusEffects {
+    active_effects: HashMap<StatusEffect, ActiveStatusEffect>,
+    new_effects: HashMap<StatusEffect, ActiveStatusEffect>,
+}
+
+impl ActiveStatusEffects {
+    pub fn add(&mut self, effect: ActiveStatusEffect) {
+        self.new_effects.insert(effect.status_effect(), effect);
+    }
+
+    pub fn active_effects(&self) -> &HashMap<StatusEffect, ActiveStatusEffect> {
+        &self.active_effects
+    }
+
+    pub fn active_effects_mut(&mut self) -> &mut HashMap<StatusEffect, ActiveStatusEffect> {
+        &mut self.active_effects
+    }
+
+    pub fn new_effects(&self) -> &HashMap<StatusEffect, ActiveStatusEffect> {
+        &self.new_effects
+    }
+
+    pub fn new_effects_mut(&mut self) -> &mut HashMap<StatusEffect, ActiveStatusEffect> {
+        &mut self.new_effects
+    }
+}
+
+/// Represents an active status effect.
+#[derive(Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash, Debug)]
+pub struct ActiveStatusEffect {
+    effect: StatusEffect,
+    amplifier: i8,
```

Comment:
Why is this an i8? This field is a byte and so cannot be negative. We can also only go up to 127 not 255 with an i8.
```suggestion
    amplifier: u8,
```

Reply:
Well Java doesn't have unsigned bytes. Afaict, the packet also doesn't use an unsigned byte. If you really believe it should be unsigned, I'll change it back